dsa-23au/java-dsa/pswish_insertion_app/src/main/java/com/pswishcorp/app/HeapSort.java

```diff
@@ -0,0 +1,58 @@
+package com.pswishcorp.app;
+
+public class HeapSort {
+    /**
+     * Method for HeapSort 
+     * @ param 1 the index of an item in the heap
+     * @ return the index of the left child 
+     * */
+    
+
+    private static int leftChild (int i) {
+        return 2 * i +1 ;
+    }
+
+    /**
+     * Method for heapsort that is used in deleteMax and buildHeap.
+     * @param a an array of Comparable items.
+     * @int i the position from which to percolate down.
+     * @int n the logical size of the binary heap.  
+     */
+
+    private static <AnyType extends Comparable<? super AnyType>> void percDown (AnyType [] a, int i, int n){
+        AnyType tmp;
+        int child;
+        for (tmp = a[i]; leftChild(i) < n; i = child){
+            child = leftChild(i);
+            if(child != n -1 && a[child].compareTo(a[child+1]) < 0){
+                child++;
+            }
+            if(tmp.compareTo(a[child])< 0){
+                a[i] = a[child];
+            }            
+            else
+                break;
+        }
+        a[i] = tmp;
+    }
+
+    /**
+     * Standard heapsort.
+     * @param a an array of Comparable items
+     */
+    public static <AnyType extends Comparable<? super AnyType>> void heapsort(AnyType [] a){
```

Comment:
The first loop is responsible for building a max heap from the input array. The process starts from the middle of the array and moves towards the front.